tensorflow_serving/util/json_tensor_test.cc

```diff
@@ -1017,6 +1017,51 @@ TEST(JsontensorTest, FromJsonMultipleNamedTensorsErrors) {
   EXPECT_THAT(status.error_message(), HasSubstr("inconsistent batch size"));
 }
 
+TEST(JsontensorTest, FromJsonSingleZeroBatchTensor) {
+  TensorMap tensormap;
+  ASSERT_TRUE(TextFormat::ParseFromString(R"(
+    dtype: DT_INT32
+    tensor_shape {
+      dim { }
+      dim { size: 2 }
+    }
+    )",
+                                          &tensormap["int_tensor"]));
+
+  string json;
+  TF_EXPECT_OK(
+      MakeJsonFromTensors(tensormap, JsonPredictRequestFormat::kRow, &json));
+  TF_EXPECT_OK(CompareJson(json, R"({ "predictions": [] })"));
+}
+
+TEST(JsontensorTest, FromJsonMultipleZeroBatchTensors) {
+  TensorMap tensormap;
+  ASSERT_TRUE(TextFormat::ParseFromString(R"(
+    dtype: DT_STRING
+    tensor_shape {
+      dim { }
+      dim { size: 1 }
+      dim { size: 2 }
+    }
+    )",
+                                          &tensormap["str_tensor"]));
+
+  ASSERT_TRUE(TextFormat::ParseFromString(R"(
+    dtype: DT_FLOAT
+    tensor_shape {
+      dim { }
+      dim { size: 1 }
```

Comment:
Nit: can you change this size 1 to a different number, to verify that we only require the first dimension to match?

Reply:
Updated in https://github.com/tensorflow/serving/pull/1615/commits/1f7aebd906a70ba0fa04105ceee6227960b764f7